server/src/main/java/org/opensearch/rest/NamedRoute.java

```diff
@@ -9,17 +9,23 @@
 package org.opensearch.rest;
 
 import org.opensearch.OpenSearchException;
+import org.opensearch.transport.TransportService;
+
+import java.util.Set;
 
 /**
  * A named Route
  *
  * @opensearch.internal
  */
 public class NamedRoute extends RestHandler.Route {
+
     private static final String VALID_ACTION_NAME_PATTERN = "^[a-zA-Z0-9:/*_]*$";
     static final int MAX_LENGTH_OF_ACTION_NAME = 250;
 
-    private final String name;
+    private String name;
+
+    private Set<String> actionNames;
```

Comment:
👎  NamedRoute should only support a single name, just like TransportActions. In addition to the actionName, when plugins convert to extensions they need support for legacy action names. This should be changed back.

Reply:
This is to allow mapping same route against different transport action names. It was modified per @peternied 's suggestion of supporting edge cases where a route could be mapped to multiple names as they have not been switched. For example, cluster permissions with word `master`.